docs/devGuide/migratingToTypeScript.md

```diff
@@ -0,0 +1,135 @@
+
+   A common error in newly renamed files comes from the way modules are imported/exported. The syntax for those can be converted according to the table below:
+
+   Case           | CommonJS                   | TypeScript equivalent                                            | ES6
+   ---------------|----------------------------|------------------------------------------------------------------|-----------------------
+   Default import | `const x = require(y)`     | `import x = require(y)`                                          | `import x from y`
+   Named import   | `const { x } = require(y)` | N/A, workaround:<br>`import z = require(y)`<br>`const { x } = z` | `import { x } from y`
+   Named export   | `module.exports = { x }`   | `export = { x }`                                                 | `export { x }`
+   Default export | `module.exports = x`       | `export = x`                                                     | `export default x`
+
+   Similar to the process of refactoring a function, you might have to **adjust how these files are imported by other TypeScript (`.ts`) files**. If you are changing it to the TypeScript equivalent syntax, then it must be imported with so as well. The same goes for the ES6 syntax. 
+
+   Start with the TypeScript equivalent syntax, as it is compiled to be exactly the same as the CommonJS syntax (except the named import case), making it the safest choice when adapting the file. You can read more about this syntax in the [official documentation](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require).
+
+   The ES6 syntax compiles into something more than just the CommonJS syntax when compiled, and might lead to unexpected behaviours when another `.js` file imports the compiled file, to the point of failing tests in some cases. You can try changing to the ES6 syntax, but only proceed if there are no failing tests. The TypeScript equivalent syntax is enough for the initial adaptation.
+
+   However, in some cases, changing to the ES6 syntax might be necessary. For instance, you cannot use the TypeScript equivalent syntax if you want to import/export TypeScript specific constructs, such as types, interfaces, and enums. Again, when doing so, make sure that all tests pass.
+
+   <box type="info">
+
+   We will perform a mass change into ES6 syntax after all files in core have been adapted into TypeScript.
+   </box>
+
+3. **Adapt the files** to fully use TypeScript.
+
+   The errors from TypeScript and `typescript-eslint` can guide you on what to fix. Only fix what is necessary and do not modify any code functionality. Avoid using `any` as best as you can.
+
+   If you happen to encounter type errors related to using names from an internal dependency that is still in JavaScript (`.js`), you can try to infer simple types in your file first as a "stand-in" of the more robust types when that internal dependency is eventually migrated.
```

Comment:
```suggestion
   If you happen to encounter type errors related to using names from another MarkBind import that is still in JavaScript (`.js`), you can try to infer simple types in your file first as a "stand-in" of the more robust types when that internal dependency is eventually migrated.
```

> related to using names

👀 what's this ("names")?

Reply:
It's my attempt to generalize referring to constants, properties, members, etc. The case that I saw that , was that a file in migration is trying to access a static class member of another file that is still in `.js`. We have to make do for now inferring simple types like `Record<string, any>` for example while we're waiting for the other `.js` file to be migrated.